extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesConfigUtil.java

```diff
@@ -0,0 +1,121 @@
+package io.quarkus.kubernetes.deployment;
+
+import static io.quarkus.kubernetes.deployment.Constants.*;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+
+import io.dekorate.utils.Strings;
+
+public class KubernetesConfigUtil {
+
+    private static final String DEKORATE_PREFIX = "dekorate.";
+    private static final String QUARKUS_PREFIX = "quarkus.";
+
+    //Most of quarkus prefixed properties are handling directly by the config items (KubenretesConfig, OpenshiftConfig, KnativeConfig)
+    //We just need group, name & version parsed here, as we don't have decorators for these (low level properties).
+    private static final Set<String> QUARKUS_PREFIX_WHITELIST = new HashSet(Arrays.asList("group", "name", "version"));
+
+    private static final Set<String> ALLOWED_GENERATORS = new HashSet(
```

Comment:
I suggest to use `<>` when creating the `HashSet` to suppress warnings about using raw types